[2025-10-30T14:50:02.342+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-30T14:50:02.363+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-30T14:45:00+00:00 [queued]>
[2025-10-30T14:50:02.370+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-30T14:45:00+00:00 [queued]>
[2025-10-30T14:50:02.371+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-10-30T14:50:02.382+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_weather_data> on 2025-10-30 14:45:00+00:00
[2025-10-30T14:50:02.388+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'flood_prediction_pipeline', 'fetch_weather_data', 'scheduled__2025-10-30T14:45:00+00:00', '--job-id', '75', '--raw', '--subdir', 'DAGS_FOLDER/flood_prediction_dag.py', '--cfg-path', '/tmp/tmpn00oyyg8']
[2025-10-30T14:50:02.388+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=3892) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-10-30T14:50:02.390+0000] {standard_task_runner.py:91} INFO - Job 75: Subtask fetch_weather_data
[2025-10-30T14:50:02.391+0000] {standard_task_runner.py:63} INFO - Started process 3895 to run task
[2025-10-30T14:50:02.429+0000] {task_command.py:426} INFO - Running <TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-30T14:45:00+00:00 [running]> on host 31ca9a0c8d51
[2025-10-30T14:50:02.492+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='flood_prediction_pipeline' AIRFLOW_CTX_TASK_ID='fetch_weather_data' AIRFLOW_CTX_EXECUTION_DATE='2025-10-30T14:45:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-30T14:45:00+00:00'
[2025-10-30T14:50:02.494+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-30T14:50:02.560+0000] {logging_mixin.py:188} INFO - âœ… Weather fetched. Rain (1h/24h): 4.72 / 4.72
[2025-10-30T14:50:02.562+0000] {python.py:237} INFO - Done. Returned value was: {'temp': 25.15, 'humidity': 71.37, 'pressure': 997.66, 'wind_speed': 17.5, 'rain_1h': 4.72, 'rain_3h': 4.72, 'rain_6h': 4.72, 'rain_24h': 4.72, 'rain_intensity': 4.72, 'temp_delta': -7.1200000000000045, 'humidity_delta': 1.2700000000000102, 'hour': 14, 'day': 30, 'month': 10, 'year': 2025}
[2025-10-30T14:50:02.564+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-30T14:50:02.584+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=flood_prediction_pipeline, task_id=fetch_weather_data, run_id=scheduled__2025-10-30T14:45:00+00:00, execution_date=20251030T144500, start_date=20251030T145002, end_date=20251030T145002
[2025-10-30T14:50:02.605+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-10-30T14:50:02.626+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-10-30T14:50:02.630+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
